### ISSUE TITLE
`isinstance-second-argument-not-valid-type` false positive for union types with `None`

### ISSUE THREAD

**Issue by ytoku** [pylint-dev/pylint#8424]
### Bug description

#8206 does not fix the issue #8205 if the union type contains `None`.
```
# pylint: disable=missing-docstring,invalid-name
a = 1
isinstance(a, None | int)  # raises W1116
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:3:0: W1116: Second argument of isinstance is not a type (isinstance-second-argument-not-valid-type)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```

### Expected behavior

no warnings

### Pylint version

```shell
pylint 2.17.0
astroid 2.15.0
Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]
```
